usbsdmux/i2c_gpio.py

```diff
@@ -97,6 +97,20 @@ def output_values(self, values: int, bitmask: int = 0xFF):
             val = val | (values & bitmask)  # set bits set in values and bitmask
             self._write_register(self.register_outputPort, val)
 
+    def register_reset_bit(self, addr: int, bitmask: int):
+        """Resets the bits with '1' in <bitmask> in the register <addr>"""
+
+        val = self.read_register(addr)[0]
+        val = (val & ~bitmask) & 0xFF
+        self._write_register(addr, val)
+
+    def register_set_bit(self, addr: int, bitmask: int):
+        """Sets the bits with '1' in <bitmask> in the register <addr>"""
+
+        val = self.read_register(addr)[0]
+        val = (val | bitmask) & 0xFF
+        self._write_register(addr, val)
+
```

Comment:
The usual primitive for this is a _update_mask function, which can perform either function.

Reply:
Not sure what you mean by this. Internet search does not report any useful results. Do you have a pointer?

Reply:
Something like https://elixir.bootlin.com/linux/latest/source/drivers/base/regmap/regmap.c#L3169 and https://elixir.bootlin.com/linux/latest/source/drivers/base/regmap/regmap.c#L3151 ...

So you have `mask` and `val` as input, read the register, reset the mask bits (`tmp = reg & ~mask`), set the bits  in `val` allowed by the mask (`tmp |= val & mask`) and write back to the register. This single function allows updating any bits in a register to new values without touching the others. It can replace your `_reset_bit` and `_set_bit` functions, while also allowing to replace specific bits with a new value consisting of ones and zeros.

Reply:
Ah, you mean like the method I had already implemented here: https://github.com/linux-automation/usbsdmux/pull/52/files#diff-11952356576c020ad5be17c7705787e2c06813dd213effeca7e1557ecef29001R89
Removed both methods. Used the existing one. Thx!